datafusion-cli/src/print_format.rs

```diff
@@ -209,14 +211,175 @@ impl PrintFormat {
         }
         Ok(())
     }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn process_batch(
```

Comment:
Given that we now have some more time, I wonder if you could potentially consider refactoring this code now. 
- https://github.com/apache/datafusion/issues/14886

I found it somewhat hard to understand when I was trying to debug why the explain plans were not showing up reasonably.

Reply:
Thank you @alamb for this suggestion, may after this PR finalizing. We can open the https://github.com/apache/datafusion/pull/14961 to my branch? Is it a common way for us to collaborate?